csharp/src/Google.Protobuf/Reflection/EnumDescriptor.cs

```diff
@@ -136,7 +136,13 @@ public EnumValueDescriptor FindValueByName(string name)
         /// </summary>
         public T GetOption<T>(Extension<EnumOptions, T> extension)
         {
-            var value = Proto.Options.GetExtension(extension);
+            var options = Proto.Options;
+            if (options == null)
+            {
+                return default(T);
```

Comment:
I don't think returning `default(T)` is a good idea. It seems that for e.g. extension of type int, you won't be able to differentiate between the option not being present at all (which should return null) and the option having value of 0.

Given that the GetOption API is relatively new (and looks like it has been kind of broken for the very beginning), should we just introduce TryGetOption methods instead, to keep the semantics clean?

the GetOption can either be removed (that should be fine because it was only introduced as part of the proto2 work and we've marked proto2 support as "experimental")  or it can just throw an exception if the value is not present (which renders it not very useful, so we might wanna mark it as obsolete?).

Reply:
Returning `default(T)` is a mistake. It should be the default value for the extension.

Reply:
Still, even if you return `extension.DefaultValue`, it is unclear to me how I would distinguish between the extension not being present at all  and it being present and having the default value.

Reply:
You can't, which is why #7282 was made. Should we also add a `HasOption` API to go with it?